.github/workflows/qcom-preflight-checks-reusable-workflow.yml

```
@@ -73,6 +73,7 @@ jobs:
           SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
 
       - name: Upload SARIF file for GitHub Advanced Security Dashboard
+        if: ${{ github.event.repository.public }}
```

`.public` isn't documented as existing in the [repository object](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository) so I think you want to use `.visibility`, which can have values of `public`, `private`, or `internal`.

And should this be part of the `if` on L63 so that none of the semgrep steps run on non-public repos?

```suggestion
        if: ${{ github.event.repository.visibility == 'public' }}
```

---

I think we still want semgrep to run on PRs when visibility is still internal (i.e. repo is being staged). It's just the bit that uploads to the security dashboard that we know won't function until it's made public and we enable the security config for the repo.

@njjetha Nasser makes a good point on using visibility since it's documented